python-frontend/typeshed_serializer/tests/test_symbols_merger.py

```diff
@@ -0,0 +1,195 @@
+
+    # Merge single module
+    merged_modules = symbols_merger.merge_modules({"ssl"}, {"37": {"ssl": ssl_module_symbol}})
+    assert len(merged_modules) == 1
+    ssl_merged_module_symbol = merged_modules["ssl"]
+
+    # Merged module symbol conversion to proto
+    ssl_merged_symbol_proto = ssl_merged_module_symbol.to_proto()
+    assert ssl_merged_symbol_proto.fully_qualified_name == ssl_merged_module_symbol.fullname
+
+    assert_merged_class_symbol_to_proto(ssl_merged_symbol_proto.classes, ssl_merged_module_symbol.classes)
+    assert_merged_function_symbol_to_proto(ssl_merged_symbol_proto.functions, ssl_merged_module_symbol.functions)
+
+
+def test_basic_functions_merge(typeshed_stdlib):
+    abc_module_symbol = symbols.ModuleSymbol(typeshed_stdlib.files.get("abc"))
+    handled_funcs = {}
+
+    # merge from empty
+    symbols_merger.merge_functions(abc_module_symbol, handled_funcs, '37')
+    assert len(handled_funcs) == len(abc_module_symbol.functions)
+    for merged_func in handled_funcs.values():
+        assert len(merged_func) == 1
+        assert merged_func[0].valid_for == ['37']
+
+    # merge with identical model
+    symbols_merger.merge_functions(abc_module_symbol, handled_funcs, '38')
+    assert len(handled_funcs) == len(abc_module_symbol.functions)
+    for merged_func in handled_funcs.values():
+        assert len(merged_func) == 1
+        assert merged_func[0].valid_for == ['37', '38']
+
+
+def test_basic_classes_merge(typeshed_stdlib):
+    abc_module_symbol = symbols.ModuleSymbol(typeshed_stdlib.files.get("abc"))
+    handled_classes = {}
+
+    # merge from empty
+    symbols_merger.merge_classes(abc_module_symbol, handled_classes, '37')
+    assert len(handled_classes) == len(abc_module_symbol.classes)
+    for merged_class in handled_classes.values():
+        assert len(merged_class) == 1
+        assert merged_class[0].valid_for == ['37']
+
+    # merge with identical model
+    symbols_merger.merge_classes(abc_module_symbol, handled_classes, '38')
+    assert len(handled_classes) == len(abc_module_symbol.classes)
+    for merged_class in handled_classes.values():
+        assert len(merged_class) == 1
+        assert merged_class[0].valid_for == ['37', '38']
+
+
+def test_overloaded_functions_merge(typeshed_stdlib):
+    ssl_module_symbol = symbols.ModuleSymbol(typeshed_stdlib.files.get("ssl"))
+    handled_overloaded_funcs = {}
+    socket_class_symbol = [c for c in ssl_module_symbol.classes if c.fullname == 'ssl.SSLSocket'][0]
+
+    # merge from empty
+    symbols_merger.merge_overloaded_functions(socket_class_symbol, handled_overloaded_funcs, '37')
+    assert len(handled_overloaded_funcs) == len(socket_class_symbol.overloaded_methods)
+    for merged_overloaded_func in handled_overloaded_funcs.values():
+        assert len(merged_overloaded_func) == 1
+        assert merged_overloaded_func[0].valid_for == ['37']
+
+    # merge with identical model
```

Comment:
I think it might be worth testing a merge when the models are not identical.
If I'm not wrong I didn't see it tested elsewhere